### PR TITLE
Testuojuos

### PR DESCRIPTION
**Pull request by vilmantas**

### CODE REVIEW

File: src/Template.API/Controllers/UserController.cs [hunk 1]
```diff
@@ -5,13 +5,15 @@
...
 using Template.Repositories.Repositories;
+using Template.Services.Services;
 
 namespace Template.API.Controllers
 {
     [Route("api/[controller]")]
     public class UserController : Controller
     {
-        private readonly IUserRepository _userRepository;
+        private readonly IUserRepository _userRepository; // why readonly?
```
Comment (vilmantas) [inline src/Template.API/Controllers/UserController.cs]: Don't leave questions like this in code. If you need information about something make a note and ask.

File: src/Template.API/Controllers/UserController.cs [hunk 2]
```diff
@@ -21,13 +23,18 @@ public UserController(IUserRepository userRepository)
         public string AddUser(string name, string surname)
         {
             _userRepository.Add(new User() { Name = name, Surname = surname });
+            //_userService.Add(name, surname);
```
Comment (vilmantas) [inline src/Template.API/Controllers/UserController.cs]: Try not to leave commented lines of code. This line should work, why didn't you use it?

File: src/Template.API/Controllers/UserController.cs [hunk 3]
```diff
@@ -21,13 +23,18 @@ public UserController(IUserRepository userRepository)
...
             return $"Added user to list: {name} {surname}";
         }
 
         [HttpGet("get/{id}")]
         public IActionResult GetUser(int id)
         {
             return Ok(_userRepository.Get(id) ?? new User() { Name = "Not", Surname = "Found" });
+           
+           // IUserService user = GetUserById(id);
```
Comment (vilmantas) [inline src/Template.API/Controllers/UserController.cs]: Same as above. This wont work tho.

File: src/Template.Common/Exceptions/ExceptionCode.cs [hunk 4]
```diff
@@ -26,7 +22,27 @@ public enum User
...
+            [Description("Product with the same name already exists")]
+            DuplicateProduct = 30001,
+
+            [Description("Cannot create product with empty name, Vendor price, Listing price or status ")]
+            EmptyField = 30002,
+
+            [Description("Product Vendor price cannot be bigger than a product listing price")]
+            VendorBiggerThanListingPrice = 30003,
+
+            [Description("Product name doesn't exists")]
```
Comment (vilmantas) [inline src/Template.Common/Exceptions/ExceptionCode.cs]: Should be something like "Product with provided name doesn't exist." or "Product with that name was not found". Current statement doesn't really make sense.

File: src/Template.Entities/DbModels/Product.cs [hunk 5]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+using Template.Entities.DbModels.Base;
+
+namespace Template.Entities.DbModels
+{
+  public  enum Activity
```
Comment (vilmantas) [inline src/Template.Entities/DbModels/Product.cs]: This field isn't about the "State" of the product. "Comming" and "None" Shouldn't be here. If we needed something like that we'd create a different field called "State". 
A bit more info here: https://softwareengineering.stackexchange.com/questions/159232/should-we-ever-delete-data-in-a-database

File: src/Template.Entities/DbModels/Product.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    //    bool Delete(IProductModel model);
+    //}
+   public class Product : BaseEntity
+    {
+       
+        public string Name { get; set; }
+        public string Description { get; set; }
+        public double VendorPrice { get; set; }
+        public double ListingPrice { get; set; }
+        public DateTime CreateDate { get; set; } //Maybe it would be better to name it just Date
```
Comment (vilmantas) [inline src/Template.Entities/DbModels/Product.cs]: "Create" is a noun to make the "Date" make more sense. Imagine if we'd have another field "LastOrderDate", then it wouldn't be clear what the "Date" field is for.

File: src/Template.API/Controllers/UserController.cs (hunk 2 above)
Comment (boom94) [reply src/Template.API/Controllers/UserController.cs]: because it was untested. I will delete all comment when I test them all.

**Pull request closed without merging**